Kernel/Coredump.h

```diff
@@ -21,7 +23,47 @@ class Coredump {
     ErrorOr<void> write();
 
 private:
-    Coredump(NonnullLockRefPtr<Process>, NonnullLockRefPtr<OpenFileDescription>);
+    class FlatRegionData {
+    public:
+        explicit FlatRegionData(Memory::Region const& region)
+        {
+            m_access = region.access();
+            m_is_executable = region.is_executable();
+            m_is_kernel = region.is_kernel();
+            m_is_readable = region.is_readable();
+            m_is_writable = region.is_writable();
+            m_name = MUST(KString::try_create(region.name()));
```

Comment:
Never allocate inside a constructor...

Reply:
I'll move it into the loop in `Coredump::try_create()`.